src/main/java/com/habicus/core/security/JWTAuthorizationFilter.java

```diff
@@ -0,0 +1,82 @@
+| |_| |/ _` | '_ \| |/ __| | | / __|
+|  _  | (_| | |_) | | (__| |_| \__ \
+|_| |_|\__,_|_.__/|_|\___|\__,_|___/
+
+ * This file is part of the Habicus Core Platform (https://github.com/Habicus/Habicus-Core).
+ * Copyright (c) 2018 Habicus Core
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, version 3.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.habicus.core.security;
+
+import static com.habicus.core.security.SecurityConstants.HEADER_STRING;
+import static com.habicus.core.security.SecurityConstants.SECRET;
+import static com.habicus.core.security.SecurityConstants.TOKEN_PREFIX;
+
+import io.jsonwebtoken.Jwts;
+import java.io.IOException;
+import java.util.ArrayList;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.web.authentication.www.BasicAuthenticationFilter;
+
+public class JWTAuthorizationFilter extends BasicAuthenticationFilter {
+
+  public JWTAuthorizationFilter(AuthenticationManager authManager) {
+    super(authManager);
+  }
+
+  @Override
+  protected void doFilterInternal(
+      HttpServletRequest req, HttpServletResponse res, FilterChain chain)
+      throws IOException, ServletException {
+    String header = req.getHeader(HEADER_STRING);
+
+    if (header == null || !header.startsWith(TOKEN_PREFIX)) {
+      chain.doFilter(req, res);
+      return;
+    }
+
+    UsernamePasswordAuthenticationToken authentication = getAuthentication(req);
+
+    SecurityContextHolder.getContext().setAuthentication(authentication);
+    chain.doFilter(req, res);
+  }
+
+  private UsernamePasswordAuthenticationToken getAuthentication(HttpServletRequest request) {
+    String token = request.getHeader(HEADER_STRING);
+    if (token != null) {
+      // parse the token.
+      String user =
+          Jwts.parser()
+              .setSigningKey(SECRET.getBytes())
+              .parseClaimsJws(token.replace(TOKEN_PREFIX, ""))
+              .getBody()
+              .getSubject();
+
+      if (user != null) {
+        return new UsernamePasswordAuthenticationToken(user, null, new ArrayList<>());
+      }
+      return null;
+    }
+    return null;
```

Comment:
Why return null here? Should we also be adding logs in this function? Throwing exceptions?